microsoft-edge/webview2/concepts/process-related-events.md

```diff
@@ -0,0 +1,636 @@
+
+Subsections below:
+* [Gather process failure details](#gather-process-failure-details)
+* [Handle unexpected exits for various types of processes](#handle-unexpected-exits-for-various-types-of-processes)
+* [Handle unresponsive renderers](#handle-unresponsive-renderers)
+
+
+<!-- ------------------------------ -->
+#### Gather process failure details
+
+
+The `ProcessFailed` event provides information about the **kind of failure** and the **reason** why it occurred. Your application can interpret these details in the following manner.
+
+Subsections below:
+* [Failure kind](#failure-kind)
+* [Failure reason](#failure-reason)
+* [Exit code, process description, and frames information](#exit-code-process-description-and-frames-information)
+
+
+<!-- ---------- -->
+###### Failure kind
+
+Due to historical reasons, _failure kind_ is a combination of process kind (browser, renderer, gpu, ...) and failure (exit, unresponsiveness). Your application can use _failure kind_ to determine the process that has failed. Renderer processes are further divided in _main frame_ renderer (`RenderProcessExited`, `RenderProcessUnresponsive`) and _subframe_ renderer (`FrameRenderProcessExited`). For more details about the conditions under which each specific _failure kind_ is used, see:
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+* [CoreWebView2ProcessFailedKind Enum](https://learn.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2processfailedkind)
+
+##### [WinRT/C#](#tab/winrtcsharp)
+
+* [CoreWebView2ProcessFailedKind Enum](https://learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2processfailedkind)
+
+##### [Win32/C++](#tab/win32cpp)
+
+* [COREWEBVIEW2_PROCESS_FAILED_KIND enum](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/webview2-idl#corewebview2_process_failed_kind)
+
+---
+
+
+<!-- ---------- -->
+###### Failure reason
+
+When the _failure kind_ is for an unexpected exit, _failure reason_ indicates the category of the problem causing the exit. Some of the _failure reasons_ are only applicable to specific _failure kinds_.
+
+For unresponsive renderer _failure kind_, the _failure reason_ is always `unresponsive`.
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+* [CoreWebView2ProcessFailedReason Enum](https://learn.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2processfailedreason)
+
+##### [WinRT/C#](#tab/winrtcsharp)
+
+* [CoreWebView2ProcessFailedReason Enum](https://learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2processfailedreason)
+
+##### [Win32/C++](#tab/win32cpp)
+
+* [COREWEBVIEW2_PROCESS_FAILED_REASON enum](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/webview2-idl#corewebview2_process_failed_reason)
+
+---
+
+
+<!-- ---------- -->
+###### Exit code, process description, and frames information
```

Comment:
It might make sense to move this section to be the top section of "Gather process failure details" and include Failure kind and failure reason in the list.

Reply:
@mikehoffms can you make this change? I'm not sure what the plans are around headings.

Reply:
I moved the section up to top of parent h2 section.  Didn't change heading or content yet.  Might move back down - discuss in meeting.  Adjust top-of-page TOC.

Reply:
The section "Gathering process failure details" essentially replaced that section.